client/ayon_core/pipeline/workfile/utils.py

```diff
@@ -124,3 +142,624 @@ def should_open_workfiles_tool_on_launch(
     if output is None:
         return default_output
     return output
+
+
+def _get_event_context_data(
+    project_name: str,
+    folder_entity: dict[str, Any],
+    task_entity: dict[str, Any],
+    host_name: str,
+):
+    return {
+        "project_name": project_name,
+        "folder_id": folder_entity["id"],
+        "folder_path": folder_entity["path"],
+        "task_id": task_entity["id"],
+        "task_name": task_entity["name"],
+        "host_name": host_name,
+    }
+
+
+def save_workfile_info(
+    project_name: str,
+    task_id: str,
+    rootless_path: str,
+    host_name: str,
+    version: Optional[int],
+    comment: Optional[str],
+    description: Optional[str],
+    username: Optional[str] = None,
+    workfile_entities: Optional[list[dict[str, Any]]] = None,
+):
+    # TODO create pipeline function for this
+    if workfile_entities is None:
+        workfile_entities = list(ayon_api.get_workfiles_info(
+            project_name,
+            task_ids=[task_id],
+        ))
+
+    workfile_entity = next(
+        (
+            _ent
+            for _ent in workfile_entities
+            if _ent["path"] == rootless_path
+        ),
+        None
+    )
+
+    if username is None:
+        username = get_ayon_username()
+
+    if not workfile_entity:
+        return _create_workfile_info_entity(
+            project_name,
+            task_id,
+            host_name,
+            rootless_path,
+            username,
+            version,
+            comment,
+            description,
+        )
+
+    data = {}
+    for key, value in (
+        ("host_name", host_name),
+        ("version", version),
+        ("comment", comment),
+    ):
+        if value is not None:
+            data[key] = value
```

Comment:
List comprehension is in this case readable too and a little bit faster.

```suggestion
    data = {
        key: value
        for key, value in (
            ("host_name", host_name),
            ("version", version),
            ("comment", comment),
        )
        if value is not None
    }
```

Reply:
Less readable to me...